File: app/src/main/java/com/github/sewerina/meter_readings/database/AppDao.java
package com.github.sewerina.meter_readings.database;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

import io.reactivex.Completable;
import io.reactivex.Single;

@Dao
public interface AppDao {

    @Query("delete from reading")
    Completable clearTableReadingRx();

    @Query("delete from home")
    Completable clearTableHomeRx();

    @Insert()
    Completable insertInTableReadingRx(List<ReadingEntity> readingEntities);

    @Insert()
    Completable insertInTableHomeRx(List<HomeEntity> homeEntities);

    @Query("select * from reading")
    List<ReadingEntity> getAllReadings();

    @Query("select * from reading")
    Single<List<ReadingEntity>> getAllReadingsRx();

    @Query("select * from reading where homeId =:homeId order by date desc")
    Single<List<ReadingEntity>> getReadingsForHomeRx(int homeId);

    @Query("select * from home")
    Single<List<HomeEntity>> getHomesRx();

    @Query("select * from home")
    LiveData<List<HomeEntity>> getHomesLiveData();

//    @Insert
//    Completable insertReadingRx(ReadingEntity entity);

    @Insert
    Single<Long> insertReadingRx(ReadingEntity entity);

    @Update
    Completable updateReadingRx(ReadingEntity entity);

    @Delete
    Completable deleteReadingRx(ReadingEntity entity);

//    @Insert
//    Completable insertHomeRx(HomeEntity entity);

    @Insert
    Single<Long> insertHomeRx(HomeEntity entity);

    @Update
    Completable updateHomeRx(HomeEntity entity);

//    @Update
//    Single<Long> updateHomeRx(HomeEntity entity);

    @Delete
    Completable deleteHomeRx(HomeEntity entity);

}

File: app/src/main/java/com/github/sewerina/meter_readings/database/AppDao.kt
package com.github.sewerina.meter_readings.database

import androidx.lifecycle.LiveData
import androidx.room.*

@Dao
interface AppDao {
    @Query("delete from reading")
    suspend fun clearTableReading()

    @Query("delete from home")
    suspend fun clearTableHome()

    @Insert
    suspend fun insertInTableReading(readingEntities: List<ReadingEntity>)

    @Insert
    suspend fun insertInTableHome(homeEntities: List<HomeEntity>)

    @Query("select * from reading")
    suspend fun allReadings(): List<ReadingEntity>

    @Query("select * from reading where homeId =:homeId order by date desc")
    suspend fun getReadingsForHome(homeId: Int): List<ReadingEntity>

    @Query("select * from home")
    suspend fun homes(): List<HomeEntity>

    @get:Query("select * from home")
    val homesLiveData: LiveData<List<HomeEntity>>

    @Insert(entity = ReadingEntity::class)
    suspend fun insertReading(entity: NewReadingEntity): Long

    @Update
    suspend fun updateReading(entity: ReadingEntity)

    @Delete
    suspend fun deleteReading(entity: ReadingEntity)

    @Insert(entity = HomeEntity::class)
    suspend fun insertHome(entity: NewHomeEntity): Long

    @Update
    suspend fun updateHome(entity: HomeEntity)

    @Delete
    suspend fun deleteHome(entity: HomeEntity)
}